### ISSUE TITLE
Prefix is removed after the entry reaches groupSeparator in currency alias and when maxlength is set to the input

### ISSUE THREAD

**Issue by sampathkumarm** [RobinHerbots/Inputmask#2504]
Find the codepen here to reproduce the issue and I am using https://cdnjs.cloudflare.com/ajax/libs/jquery.inputmask/5.0.4/inputmask.min.js

https://codepen.io/msampathkumar/pen/WNRBLxb

Good until first 3 digits
![image](https://user-images.githubusercontent.com/6489309/116614378-b8244680-a8ee-11eb-974a-f7f508a67e07.png)

Breaks when 4th digit is entered
![image](https://user-images.githubusercontent.com/6489309/116614458-d0946100-a8ee-11eb-9107-84d2fc79ca88.png)

**Comment by gabriel-lima96** [comment]
Same here.

I came up with a workaround:
```javascript
{
  preValidation: function(buffer) {
    return buffer.length < yourmaxlength;
  }
}
```

**Comment by sampathkumarm** [comment]
> Same here.
> 
> I came up with a workaround:
> 
> ```js
> {
>   preValidation: function(buffer) {
>     return buffer.length < yourmaxlength;
>   }
> }
> ```

Didn't work for me, updated the same codepen.

**Comment by gabriel-lima96** [comment]
> Didn't work for me, updated the same codepen.

Sorry, i forgot to tell you: you have to remove maxlength from the html. I do it on the runtime.
Paste this on your codepen:
```javascript
$(document).ready(function () {
  const $currency = $("#currency");
  const maxlength = $currency.attr('maxlength') || 256;
  $currency.attr('maxlength', null);
  
  $currency.inputmask({
    alias: "currency",
    prefix: "$",
    digits: 0,
    placeholder: "",
    preValidation: function (buffer) {
      return buffer.length < maxlength;
    }
  });
});
```

**Comment by sampathkumarm** [comment]
> > Didn't work for me, updated the same codepen.
> 
> Sorry, i forgot to tell you: you have to remove maxlength from the html. I do it on the runtime.
> Paste this on your codepen:
> 
> ```js
> $(document).ready(function () {
>   const $currency = $("#currency");
>   const maxlength = $currency.attr('maxlength') || 256;
>   $currency.attr('maxlength', null);
>   
>   $currency.inputmask({
>     alias: "currency",
>     prefix: "$",
>     digits: 0,
>     placeholder: "",
>     preValidation: function (buffer) {
>       return buffer.length < maxlength;
>     }
>   });
> });
> ```

Perfect, that did the trick. Thank you so much.
But is there any ETA to get this fixed in the library itself of would this be the way to go forward.

**Comment by gabriel-lima96** [comment]
I don't think this issue should be closed. This is still a bug, i just found a workaround.

**Comment by sampathkumarm** [comment]
Re-opening as per @gabriel-lima96, thank you Gabriel.

**Comment by RobinHerbots** [comment]
@sampathkumarm , @gabriel-lima96 ,

Why didn't yoiu just try with the latest version?

**Comment by gabriel-lima96** [comment]
> @sampathkumarm , @gabriel-lima96 ,
> 
> Why didn't yoiu just try with the latest version?

Didn't we?
Just to clarify, @sampathkumarm codepen is already fixed with the workaround. You have to remove the preValidation and add maxlength on the input.

**Comment by RobinHerbots** [comment]
@gabriel-lima96,

Fixed in version 5.0.6-beta.62
